plugins/inputs/icinga2/icinga2.go

```diff
@@ -110,6 +122,66 @@ func (i *Icinga2) createHTTPClient() (*http.Client, error) {
 	return client, nil
 }
 
+func (i *Icinga2) IcingaRequest(address string) (*http.Response, error) {
+	req, err := http.NewRequest("GET", address, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if i.Username != "" {
+		req.SetBasicAuth(i.Username, i.Password)
+	}
+
+	resp, err := i.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (i *Icinga2) ParseCIBResponse(resp *http.Response) (map[string]interface{}, error) {
+	result := ResultCIB{}
+
+	err := json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if len(result.Results) == 0 {
+		return nil, fmt.Errorf("No results in Icinga2 API response")
```

Comment:
Errors in Golang need to start with a lowercase letter and you don't have arguments so maybe
```suggestion
		return nil, errors.New("no results in Icinga2 API response")
```
Same for other occurrences below.

Reply:
Fixed.